src/main/java/dev/mja00/villagerLobotimizer/LobotomizeStorage.java

```diff
@@ -237,20 +243,186 @@ private boolean canMoveCardinally(World w, int x, int y, int z, boolean roof) {
     }
 
     public final class ActivatorTask implements Runnable {
-        public ActivatorTask() {
+        public void run() {
+            // Create a copy to avoid concurrent modification
+            Set<Villager> toRemove = new HashSet<>();
+            for (Villager villager : inactiveVillagers) {
+                if (processVillager(villager, false)) {
+                    toRemove.add(villager);
+                }
+            }
+            inactiveVillagers.removeAll(toRemove);
         }
+    }
 
+    public final class DeactivatorTask implements Runnable {
         public void run() {
-            LobotomizeStorage.this.inactiveVillagers.removeIf((villager) -> LobotomizeStorage.this.processVillager(villager, false));
+            // Create a copy to avoid concurrent modification
+            Set<Villager> toRemove = new HashSet<>();
+            for (Villager villager : activeVillagers) {
+                if (processVillager(villager, true)) {
+                    toRemove.add(villager);
+                }
+            }
+            activeVillagers.removeAll(toRemove);
         }
     }
 
-    public final class DeactivatorTask implements Runnable {
-        public DeactivatorTask() {
+    public void handleBlockChange(Block block) {
```

Comment:
Technically this'd break the `/lobotomy wakeup` command as the moment someone changed blocks in that chunk it'd reload. Since we support nametags now I'd say we can just remove the wakeup command.